src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,149 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> tail;
+    private Node<T> head;
+    private int size;
+
     @Override
     public void add(T value) {
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
+        findIndexForAdd(index);
```

Comment:
this method doesnt find anything it only check if index is valid. Rename it something like "validateIndexForAdd"